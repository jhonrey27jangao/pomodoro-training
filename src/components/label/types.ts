export type LabelProps = {
  theme?: "light" | "dark";
  weight?: "bold" | "light";
  center?: boolean;
  type: "h1" | "h2" | "h3" | "h4" | "p";
  children?: any;
  strike?: boolean;
  pointer?: boolean;
  onClick?: () => void;
};
