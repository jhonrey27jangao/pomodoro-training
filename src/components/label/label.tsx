import React from "react";
import {
  StyledHeading1,
  StyledHeading2,
  StyledHeading3,
  StyledHeading4,
  StyledLabel
} from "./styles";
import { LabelProps } from "./types";

export const Label: React.FC<LabelProps> = ({
  type,
  theme,
  weight,
  center,
  children,
  strike,
  pointer,
  onClick,
}) => (
  <>
    {type === "h1" ? (
      <StyledHeading1 type={type} theme={theme} weight={weight} center={center} strike={strike} >
        {children}
      </StyledHeading1>
    ) : type === "h2" ? (
      <StyledHeading2 type={type} theme={theme} weight={weight} center={center} strike={strike} >
        {children}
      </StyledHeading2>
    ) : type === "h3" ? (
      <StyledHeading3 type={type} theme={theme} weight={weight} center={center} strike={strike}>
        {children}
      </StyledHeading3>
    ) : type === "h4" ? (
      <StyledHeading4 type={type} theme={theme} weight={weight} center={center} strike={strike} >
        {children}
      </StyledHeading4>
    ) : (
      <StyledLabel type={type} theme={theme} weight={weight} center={center} strike={strike} pointer={pointer} onClick={onClick}>
        {children}
      </StyledLabel>
    )}
  </>
);
