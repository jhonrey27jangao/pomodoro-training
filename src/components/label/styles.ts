import styled from "styled-components";
import { LabelProps } from "./types";

export const StyledHeading1 = styled.h1<LabelProps>`
  font-size: 50px;
  vertical-align: baseline;
  font-weight: ${(props) => (props.weight === "bold" ? "700" : "100")};
  text-align: ${(props) => props.align};
  text-decoration: ${(props) => (props.strike ? "line-through" : "none")};
`;

export const StyledHeading2 = styled.h2<LabelProps>`
  font-size: 30px;
  font-family: inherit;
  vertical-align: baseline;
  color: ${(props) => (props.theme === "light" ? "#fff" : "#7a7d86")};
  font-weight: ${(props) => (props.weight === "bold" ? "900" : "100")};
  text-align: ${(props) => props.align};
  text-decoration: ${(props) => (props.strike ? "line-through" : "none")};
`;

export const StyledHeading3 = styled.h3<LabelProps>`
  font-size: 21px;
  font-family: inherit;
  vertical-align: baseline;
  color: ${(props) => (props.theme === "light" ? "#fff" : "#7a7d86")};
  font-weight: ${(props) => (props.weight === "bold" ? "900" : "100")};
  text-align: ${(props) => props.align};
  text-decoration: ${(props) => (props.strike ? "line-through" : "none")};
`;

export const StyledHeading4 = styled.h4<LabelProps>`
  font-family: inherit;
  vertical-align: baseline;
  color: ${(props) => (props.theme === "light" ? "#fff" : "#7a7d86")};
  font-weight: ${(props) => (props.weight === "bold" ? "900" : "100")};
  text-align: ${(props) => props.align};
  text-decoration: ${(props) => (props.strike ? "line-through" : "none")};
`;

export const StyledLabel = styled.p<LabelProps>`
  display: inline;
  font-family: inherit;
  vertical-align: baseline;
  color: ${(props) => (props.theme === "light" ? "#fff" : "#7a7d86")};
  font-weight: ${(props) => (props.weight === "bold" ? "900" : "100")};
  text-align: ${(props) => props.align};
  text-decoration: ${(props) => (props.strike ? "line-through" : "none")};
  cursor: ${(props) => (props.pointer ? "pointer" : "none")};
`;
