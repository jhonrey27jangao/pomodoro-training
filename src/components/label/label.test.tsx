import React from "react";
import { render } from "@testing-library/react";
import { Label } from "./label";

describe("Test for Label Component", () => {
  it("should render Label with type of p", () => {
    const { container } = render(<Label type="p" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Label with type of h1", () => {
    const { container } = render(<Label type="h1" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Label with type of h2", () => {
    const { container } = render(<Label type="h2" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Label with type of h3", () => {
    const { container } = render(<Label type="h3" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Label with type of h4", () => {
    const { container } = render(<Label type="h4" />);
    expect(container).toMatchSnapshot();
  });
});
