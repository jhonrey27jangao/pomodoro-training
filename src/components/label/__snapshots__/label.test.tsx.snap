// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Label Component should render Label 1`] = `
<div>
  <p
    class="sc-AxhUy gaWmhW"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h1 1`] = `
<div>
  <h1
    class="sc-AxjAm kLSFdP"
    type="h1"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h1 2`] = `
<div>
  <h1
    class="sc-AxjAm kLSFdP"
    type="h1"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h1 3`] = `
<div>
  <h1
    class="sc-AxjAm kLSFdP"
    type="h1"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h2 1`] = `
<div>
  <h2
    class="sc-AxirZ ebnjYA"
    type="h2"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h3 1`] = `
<div>
  <h3
    class="sc-AxiKw dMAvIU"
    type="h3"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of h4 1`] = `
<div>
  <h4
    class="sc-AxhCb AkssY"
    type="h4"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of p 1`] = `
<div>
  <p
    class="sc-AxhUy gaWmhW"
    type="p"
  />
</div>
`;

exports[`Test for Label Component should render Label with type of p 2`] = `
<div>
  <p
    class="sc-AxhUy gaWmhW"
    type="p"
  />
</div>
`;
